Documentation/StyleGuide/ContentNaming.md

```diff
@@ -0,0 +1,30 @@
+**[Table of Contents: Content Style Guide](../ContentStyleGuide.md)**
+* [Introduction](../ContentStyleGuide.md)
+* [Content Placement](ContentPlacement.md)
+* [Content Naming](ContentNaming.md)
+* [Common Layout Description (All Resources)](CommonLayout.md) 
+* [Styling Rules for Content Type](StylingContentOverview.md)
+* [Understanding Metadata](Metadata.md)
```

Comment:
This seems to be duplicated across a bunch of files.  Is there a way to avoid that?  Can we just put in a single linke to TOC?

Reply:
> This seems to be duplicated across a bunch of files. Is there a way to avoid that? Can we just put in a single linke to TOC?

Single link - wont that be boring? :)
Any better way to just show the TOC in each file? I want to know how to create a TOC that can expand and collapse nested entries..

Reply:
> This seems to be duplicated across a bunch of files. Is there a way to avoid that? Can we just put in a single linke to TOC?

I do hate the duplication...but I hate not having a TOC even more..

Reply:
> Any better way to just show the TOC in each file? I want to know how to create a TOC that can expand and collapse nested entries..

Markdown does not directly support includes but you could write simple scripts to generate the final *.md files.  Otherwise, this TOC is going to be risky or painful to edit.

Reply:
> > Any better way to just show the TOC in each file? I want to know how to create a TOC that can expand and collapse nested entries..
> 
> Markdown does not directly support includes but you could write simple scripts to generate the final *.md files. Otherwise, this TOC is going to be risky or painful to edit.

I am all for it if you can help me do it or show me a quick and easy path to writing github scripts :) Of course, my thanks to you in advance :-)

Reply:
> I am all for it if you can help me do it or show me a quick and easy path to writing github scripts :) Of course, my thanks to you in advance :-)

No, this would have to be done locally, not directly on github.  But with preview plugins like "Markdown Viewer", you can effectively create end edit `*.md` files and avoid creating dozens of little commits by trying to edit files directly on GitHub.

Reply:
@rinkug  I went through the Style Guide, it looks very good and I think it helped me fix the Personal Kanban curated content. Look it over and see what you think. This is great. 
I do agree with David that Content Placement doesn't belong. It would be nice to see how it will look when published.

Reply:
> @rinkug I went through the Style Guide, it looks very good and I think it helped me fix the Personal Kanban curated content. Look it over and see what you think. This is great.
> I do agree with David that Content Placement doesn't belong. It would be nice to see how content will actually look when published.

Pat, Glad it was useful. I added an example of an existing table for a web article. I think it will help you to to complete the kanban web article table. Please see comment in your PR #507

Reply:
> No, this would have to be done locally, not directly on github. But with preview plugins like "Markdown Viewer", you can effectively create end edit `*.md` files and avoid creating dozens of little commits by trying to edit files directly on GitHub.

@bartlettroscoe : I took your comments and changed the TOC to eliminate/minimize duplication. Please take a look

Reply:
@rinkug, this is better.  One issue is that people might get thrown off by having "Table of Contents" at the top of very page.  People may not scroll down to see the main content on each page.

Have you considered switching to the GitHub Wiki where you can have a proper sidebar TOC?  I can see examples of that in the following GitHub sites:

* https://github.com/trilinos/Trilinos/wiki
* https://github.com/kokkos/kokkos/wiki

The GitHub Wiki is actually its own GitHub repo formatted in Markdown.

Reply:
> The GitHub Wiki is actually its own GitHub repo formatted in Markdown.

@bartlettroscoe The GitHub wiki is a good option. What about readthedocs with git backend (https://docs.readthedocs.io/en/stable/)

Reply:
I like the wiki idea.  It provides a separation between operational files for the site and documentation.

Reply:
I just realized that you can't do PRs on the wiki repo.  Given the way we operate, that would seem to be something of a disadvantage.  Nevertheless, I still think the wiki is ultimately the more appropriate place to *publish* the documentation.  

According to Google, this question has come up before.  The most practical solution seems to be making a separate repo for "wiki development" and using a CI workflow to sync the development repo to the actual wiki repo.  See <https://www.growingwiththeweb.com/2016/07/enabling-pull-requests-on-github-wikis.html>

Reply:
Thanks for the addition of the web article. 
I also think the wiki might be a good idea.